builder/simplify_exprs.cc

```diff
@@ -681,6 +686,8 @@ expr simplify(const less* op, expr a, expr b) {
       r.rewrite(select(x, y, z) < z, select(x, y < z, false)) ||
       r.rewrite(y < select(x, y, w), select(x, false, y < w)) ||
       r.rewrite(w < select(x, y, w), select(x, w < y, false)) ||
+      r.rewrite(select(x, y, z) < select(x, w, u), select(x, y < w, z < u)) ||
+      r.rewrite((x < y) < (x < (y + c0)), false, eval(c0 < 0)) ||
```

Comment:
I think we should get rid of this one, #337 added rules that break these apart in a more general way (the rules immediately below), and I have another branch that adds the necessary rules for the results of that.